forest/jasmine/sogp_gps.py

```diff
@@ -1,373 +1,738 @@
+"""This module implements the sparse online gaussian process
+    algorithm for GPS data.
+    The algorithm is based on the paper [Csato and Opper (2002)]
+    and the code is based on the matlab code from the author of the paper.
+    ref: http://www.cs.ubbcluj.ro/~csatol/SOGP/thesis/Gaussian_Process.html
+    ref: http://www.cs.ubbcluj.ro/~csatol/SOGP/thesis/Sparsity_in.html
+    The algorithm is used to select basis vectors from GPS data,
+    and the selected basis vectors are used to construct the mobility matrix.
+"""
 import logging
 import math
+from typing import Dict, Tuple
+
 import numpy as np
 
 
 logging.basicConfig(level=logging.INFO)
 logger = logging.getLogger()
 
-## the radius of the earth
+# the radius of the earth
```

Comment:
Since this constant is used in a several places in Forest can we move it to `forest/constants.py` and rename to something like `EARTH_RADIUS` for clarity? Also, add to the comment that the value is in meters?